### PR TITLE
Add the IoT.js function list parser tool.

### PR DESCRIPTION
**Pull request by knightburton**
This is a javascript command line tool which can be used with node.
The script requires an IoT.js project clone on the local drive to acces the
API documentation.

IoT.jsCode-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

### CODE REVIEW

**Comment by knightburton** [comment]
@daeyeon This is the tool what I made to generate out the iotjs function list json file from the iotjs API documentation. This can be usefule if we want to change something in the list or modify it. This was the fastest and easiest way to generate out the list. I am very curious about your opinion.

File: tools/iotjs-docs-parser.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+const path = require('path');
+const fs = require('fs');
+
+/**
+ * Constant options.
+ */
+const config = {
+  output: {
+    name: 'iotjs-functions.json',
+    path: '../src/app/constants',
```
Comment (daeyeon) [inline tools/iotjs-docs-parser.js]: Shouldn't it be 'src/app/constants'?

**Comment by daeyeon** [comment]
@knightburton I agree that this tool will be useful as you pointed. One concern is if anyone has a concern about a utility based on node.js. Because we've never written such a tool so far.

File: tools/iotjs-docs-parser.js (hunk 1 above)
Comment (knightburton) [reply tools/iotjs-docs-parser.js]: Yes, it should be.

**Comment by knightburton** [comment]
@daeyeon I understand your concern. What are you suggest? Should I rewrite this in other language (like python) or should I use the IoT.js?

**Comment by yichoi** [comment]
Let's go with this and see what happens.

**Pull request merged**